app/src/main/java/com/quran/labs/androidquran/ui/preference/DataListPreference.java

```diff
@@ -0,0 +1,127 @@
+package com.quran.labs.androidquran.ui.preference;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.graphics.Color;
+import android.os.Build;
+import android.preference.ListPreference;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.CheckedTextView;
+import android.widget.ListAdapter;
+import android.widget.TextView;
+
+import com.quran.labs.androidquran.R;
+import com.quran.labs.androidquran.util.QuranSettings;
+import com.quran.labs.androidquran.util.StorageUtils;
+
+import java.util.List;
+
+public class DataListPreference extends ListPreference {
+  private CharSequence[] mFreeSpaceAmounts;
+
+  @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+  public DataListPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
```

Comment:
Regarding the proxying (point 3.), would not this cause a failure on older platforms where the super has no three-argument constructor?